compiler/modules/front/const.bal

```diff
@@ -26,11 +26,14 @@ function resolveConstDefn(ModuleSymbols mod, s:ConstDefn defn) returns s:Resolve
     }
 }
 
-function resolveConstExpr(ModuleSymbols mod, s:ModuleLevelDefn defn, s:Expr expr, t:SemType? expectedType) returns s:ResolvedConst|ResolveTypeError {
+function resolveConstExpr(ModuleSymbols mod, s:ModuleLevelDefn defn, s:Expr expr, t:SemType? expectedType, boolean td = false) returns s:ResolvedConst|ResolveTypeError {
     ExprContext cx = new ExprContext(mod, defn, constCode, constEnvironment, ());
     var { result } = check codeGenExpr(cx, constBasicBlock, expectedType, expr);
     bir:ConstOperand operand = <bir:ConstOperand>result;
     if expectedType != () && !t:isSubtype(mod.tc, operand.semType, expectedType) {
+        if td && expr is s:VarRefExpr {
+            return err:semantic(`initializer ${expr.name} is not an int`, s:locationInDefn(defn, expr.startPos));
```

Comment:
Don't understand this at all. How do you know that `int` is expected here?

Reply:
`td` is true only with `resolveConstIntExpr` so it will be an int, maybe we could propagate the type/ error message and use that here instead?